src/tardev-snapshotter/src/snapshotter.rs

```diff
@@ -1008,15 +1013,77 @@ impl TarDevSnapshotter {
                     }
                 }
 
-                trace!("Appending index to {:?}", &base_name);
-                file.rewind().context("failed to rewind the file handle")?;
-                tarindex::append_index(&mut file).context("failed to append tar index")?;
+                // Create an erofs metadata using mkfs.erofs
```

Comment:
Are you sure you dont need to close/flush the existing file before going further?

Reply:
Will flush